ui/src/track-layout/track-layout-react-utils.tsx

```diff
@@ -205,6 +206,32 @@ export function useLocationTrackStartAndEnd(
     );
 }
 
+export function useLocationTrackSwitchesAtEnds(
+    locationTrack: LayoutLocationTrack | undefined,
+    publishType: PublishType,
+    changeTime: TimeStamp,
+): { start: LayoutSwitch | undefined; end: LayoutSwitch | undefined } | undefined {
+    const id = locationTrack?.id;
+    const [switchIds, status] = useLoaderWithStatus(
+        () =>
+            id === undefined
+                ? undefined
+                : getLocationTrackSwitchesAtEnds(id, publishType, changeTime),
+        [id, publishType, changeTime],
+    );
+    const start = useSwitch(
+        switchIds?.start ?? locationTrack?.topologyStartSwitch?.switchId,
```

Comment:
Luulisin että tämä logiikka on oikein päin mutta mietittekö asian @ksurakka-solita kanssa läpi? Lähinnä mietin että jos molemmat on olemassa, pitäisikö näyttää vaihde joka alkaa raiteen loppukohdasta vai vaihde joka on vielä raiteen puolella?